### PR TITLE
Adjust Icon Path

### PR DESCRIPTION
**Pull request by adi-wtag**
Jira : https://wearewondrous.atlassian.net/browse/TSM-510

## Background
Go to informatec home page : https://www.dev-54ta5gq-no6ziiebbw5ky.eu-2.platformsh.site/en . You will notice several icons on about us section. Now go to edit page : https://www.dev-54ta5gq-no6ziiebbw5ky.eu-2.platformsh.site/en/node/1391/edit
The icons are not available in the drop down boxes. And the field is also required. So, you can't submit the form without select an icon and the proper icons are missing from the drop down box. Hence, no changes can be made on informatec home page. 

Initially, we thought that It's an informatec specific issue. But we didn't find anything. So, we focused on `icon_select_field` module. And identified that `icon_select_field` module is only showing the `svg` files that are put in this directory : `/themes/informatec/svg/`
Settings : https://www.dev-54ta5gq-no6ziiebbw5ky.eu-2.platformsh.site/en/admin/config/content/icon_select_field
But if you have any sub folders on `svg` folder then icon_select_field doesn't show those icons on the drop down list. 

## Resolution
Adjust necessary functions so that the module keep track of sub folder as well. If that works, release a major version for `icon_select_field`

# HOW TO TEST
As we identified the problem on informatec, so we are using informatec project as a testing platform.
We also locally tested this branch on informatec and integra. Icon select field seems to be working fine. We also created a draft PR to verify the `icon_select field` functionality. 
Informatec Test PR : https://github.com/wearewondrous/informatec/pull/26 

Relevant Links : 
- Home Page View : https://www.pr-26-yhpoeoi-no6ziiebbw5ky.eu-2.platformsh.site/en
- Home Page Edit : https://www.pr-26-yhpoeoi-no6ziiebbw5ky.eu-2.platformsh.site/en/node/1391/edit
- Icon Settings : https://www.pr-26-yhpoeoi-no6ziiebbw5ky.eu-2.platformsh.site/en/admin/config/content/icon_select_field

Check Icon Drop Down List: https://www.pr-26-yhpoeoi-no6ziiebbw5ky.eu-2.platformsh.site/en/admin/structure/paragraphs_type/block_icon_counter/fields/paragraph.block_icon_counter.field_icon_select

### CODE REVIEW

File: src/IconsManager.php [hunk 1]
```diff
@@ -49,7 +49,11 @@ public function scanIconsList($relative_path) {
         ->files()
         ->in($full_path);
       foreach ($finder as $file) {
+        $filePath = $file->getPath();
+        $splits = explode($full_path, $filePath);
+        $subFolder = count($splits) > 1 ? $splits[1] : '';
         $key = $file->getBasename('.' . $file->getExtension());
+        $key = $subFolder ? $subFolder . '/' . $key : $key;
```
Comment (luxflux) [inline src/IconsManager.php:56]: This supports only one additional subfolder level - could we make it recursive to support more than one level?

**Review by luxflux — CHANGES_REQUESTED** [review]

File: src/IconsManager.php (hunk 1 above)
Comment (adi-wtag) [reply src/IconsManager.php:56]:
```
$fullPath = "/var/www/drupalvm/web/themes/informatec/svg/";
$filePath = "/var/www/drupalvm/web/themes/informatec/svg/level1/level2/file.svg";
$splits = explode($fullPath, $filePath);
$subFolder = count($splits) > 1 ? $splits[1] : '';
echo $subFolder; //Prints level1/level2/file.svg
```
@luxflux bhai, It works for more than one level. I've also tested it locally.

Comment (luxflux) [reply src/IconsManager.php:56]: 🤔 Could you outline how the result (`$this->availableIcons`) looks like if I've got the following files in `$finder`?
```
[
  '/var/www/drupalvm/web/themes/informatec/svg/file.svg',
  '/var/www/drupalvm/web/themes/informatec/svg/level1/level2/file.svg', // attention, same name as `[0]`!
  '/var/www/drupalvm/web/themes/informatec/svg/level1/other-file.svg',
]
```

Comment (adi-wtag) [reply src/IconsManager.php:56]: Available icons will enlist all the icons in the svg folder including sub folders. If you go few lines down, you will find 
```
public function match($allowed_icons) {
    return ($allowed_icons) ? array_intersect_key($allowed_icons, $this->availableIcons) : $this->availableIcons;
  }
  ```
  We are actually matching the allowed icons here.
  If I configure the folder structure just as you said and dumped the allowed icons : 
Config : 
 ```
 file|RootFile
level1/level2/file|LeveledFile
level1/other-file|OtherFile
```

Dump Result : 
```
/var/www/drupalvm/web/modules/contrib/icon_select_field/src/IconsManager.php:70:
array (size=3)
  'file' => string 'RootFile' (length=8)
  'level1/level2/file' => string 'LeveledFile' (length=11)
  'level1/other-file' => string 'OtherFile' (length=9)
```
I think the attached image will describe the scenario better. 
icon_select

Comment (luxflux) [reply src/IconsManager.php:56]: Thanks for explaining 😄

**Review by luxflux — APPROVED** [review]

**Review by zar-wtag — APPROVED** [review]
Looks good.

**Review by bia-wtag — APPROVED** [review]
Checked the PR and explanation in the comments. LGTM.

**Review by isa-wtag — APPROVED** [review]

**Review by Wickeey — APPROVED** [review]

**Pull request merged**